app/controllers/application_controller.rb

```diff
@@ -55,4 +55,12 @@ def after_sign_in_path_for(_resource)
       user_root_path
     end
   end
+
+  def destroy_stale_states(jwt_id)
+    RegistrationState.where(jwt_id: jwt_id).update_all(jwt_id: nil)
+    LoginState.where(jwt_id: jwt_id).update_all(jwt_id: nil)
+
+    RegistrationState.where(jwt_id: jwt_id).destroy_all
+    LoginState.where(jwt_id: jwt_id).destroy_all
```

Comment:
This'll need to be something like

```ruby
registration_states = RegistrationState.where(jwt_id: jwt_id).pluck(:id)
login_states = LoginState.where(jwt_id: jwt_id).pluck(:id)

RegistrationState.where(id: registration_states).update_all(jwt_id: nil)
LoginState.where(id: login_states).update_all(jwt_id: nil)

RegistrationState.where(id: registration_states).destroy_all
LoginState.where(id: login_states).destroy_all
```

Reply:
Of course!  A day of whack-a-mole has destroyed all the functioning parts of my brain!